device/transport/amqp/devdoc/amqp_twin_client_requirements.md

```diff
@@ -3,106 +3,139 @@
+**SRS_NODE_DEVICE_AMQP_TWIN_16_029: [** The `enableTwinDesiredPropertiesUpdates` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler and until a message with the same `correlationId` as the one that was sent is received. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_025: [** When the `error` event is emitted, the first parameter shall be an error object obtained via the amqp `translateError` module. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_030: [** The `enableTwinDesiredPropertiesUpdates` method shall call its callback with no argument when a response is received **]**
+// TODO: we need to check error codes!
 
-### attach(callback)
+### disableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
 
-**SRS_NODE_DEVICE_AMQP_TWIN_16_001: [** The `attach` method shall attach both sender and receiver links and calls its `callback` with no argument if successful. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_031: [** The `disableTwinDesiredPropertiesUpdates` method shall call its callback immediately and with no arguments if the links are detached. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_16_002: [** The `attach` method shall call its `callback` with an `Error` if attaching either link fails. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_032: [** The `disableTwinDesiredPropertiesUpdates` method shall send an `AmqpMessage` using the `SenderLink.send` method with the following annotations and properties:
+- `operation` annotation set to `DELETE`.
+- `resource` annotation set to `/notifications/twin/properties/desired`
+- `correlationId` property set to a random integer
+- `body` set to `undefined`. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_16_003: [** The `attach` method shall call its `callback` immediately if the links are already attached. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_033: [** If the `SenderLink.send` call fails the `disableTwinDesiredPropertiesUpdates` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_16_007: [** The `attach` method shall call the `getDeviceCredentials` method on the `authenticationProvider` object passed as an argument to the constructor to retrieve the device id. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_034: [** The `disableTwinDesiredPropertiesUpdates` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler and until a message with the same `correlationId` as the one that was sent is received. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_16_008: [** The `attach` method shall call its callback with an error if the call to `getDeviceCredentials` fails with an error. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_035: [** The `disableTwinDesiredPropertiesUpdates` method shall call its callback with no argument when a response is received **]**
+// TODO: we need to check error codes!
 
-### detach(callback)
+### detach(callback: (err?: Error) => void): void;
```

Comment:
>detach [](start = 4, length = 6)

possibly silly question: should a clean detach do a DELETE operation on /notifications/twin/properties/desired like we're being told to do for MQTT?

Reply:
OK, it looks like the code does this, and I see 16_005 now.   Oops.

---
In reply to: [171684486](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171684486) [](ancestors = 171684486)